core/src/main/java/org/apache/iceberg/util/JsonUtil.java

```diff
@@ -98,6 +98,11 @@ public static <T> T parse(String json, FromJson<T> parser) {
     }
   }
 
+  public static JsonNode get(String property, JsonNode node) {
+    Preconditions.checkArgument(node.has(property), "Cannot parse missing object '%s'", property);
```

Comment:
Question: Why not use `hasNonNull`? If anything I'd make a method for getting null-able and non-null.

Reply:
that's a good point. I updated this to use `hasNonNull`